### PR TITLE
chore(api!): Replace GetRouter with AddRoute in 02-client keeper.

### PR DESCRIPTION
**Pull request by bznein**
## Description

I'm not entirely sure if I need to change some other documentation files... @crodriguezvega can you help here? :)
closes: #6926

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/02-client/keeper/keeper.go [hunk 1]
```diff
@@ -58,9 +58,8 @@ func (Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", "x/"+exported.ModuleName+"/"+types.SubModuleName)
 }
 
-// GetRouter returns the light client module router.
-func (k *Keeper) GetRouter() *types.Router {
-	return k.router
+func (k *Keeper) AddRoute(clientType string, module exported.LightClientModule) {
```
Comment (crodriguezvega) [inline modules/core/02-client/keeper/keeper.go:61]: Should we add a godoc string?

File: CHANGELOG.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (core/types) [\#6794](https://github.com/cosmos/ibc-go/pull/6794) The composite interface `QueryServer` has been removed from package `core/types`. Please use the granular `QueryServer` interfaces provided by each core submodule.
 * (light-clients/06-solomachine) [\#6888](https://github.com/cosmos/ibc-go/pull/6888) Remove `TypeClientMisbehaviour` constant and the `Type` method on `Misbehaviour`.
 * (light-clients/06-solomachine, light-clients/07-tendermint) [\#6891](https://github.com/cosmos/ibc-go/pull/6891) The `VerifyMembership` and `VerifyNonMembership` functions of solomachine's `ClientState` have been made private. The `VerifyMembership`, `VerifyNonMembership`, `GetTimestampAtHeight`, `Status` and `Initialize` functions of tendermint's `ClientState` have been made private.
-
+ * (core/02-client) [\6934](https://github.com/cosmos/ibc-go/pull/6934) The `GetRouter` method has been removed from the keeper. `AddRoute` method has been added as a wrapper that replaces `GetRouter().AddRoute()`
```
Comment (crodriguezvega) [inline CHANGELOG.md:70]: We don't need the changelog because the `GetRouter` function has not been released yet.

**Review by crodriguezvega — COMMENTED** [review]
Thank you, @bznein. I think we can remove the changelog; the update of the docs looks good!

File: CHANGELOG.md [hunk 3]
```diff
@@ -67,7 +67,6 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (core/types) [\#6794](https://github.com/cosmos/ibc-go/pull/6794) The composite interface `QueryServer` has been removed from package `core/types`. Please use the granular `QueryServer` interfaces provided by each core submodule.
 * (light-clients/06-solomachine) [\#6888](https://github.com/cosmos/ibc-go/pull/6888) Remove `TypeClientMisbehaviour` constant and the `Type` method on `Misbehaviour`.
 * (light-clients/06-solomachine, light-clients/07-tendermint) [\#6891](https://github.com/cosmos/ibc-go/pull/6891) The `VerifyMembership` and `VerifyNonMembership` functions of solomachine's `ClientState` have been made private. The `VerifyMembership`, `VerifyNonMembership`, `GetTimestampAtHeight`, `Status` and `Initialize` functions of tendermint's `ClientState` have been made private.
-
```
[the remark this answers is not in the document]
Comment (bznein) [reply CHANGELOG.md:70]: Done!

**Review by damiannolan — APPROVED** [review]

**Pull request merged**